### ISSUE TITLE
bug: Patching of model costs for past traces does not work

### ISSUE THREAD

**Issue by atrik-das-razer** [langfuse/langfuse#2702]
### Describe the bug

I am running the exact same commands as mentioned in https://langfuse.com/docs/deployment/self-host#migrate-models to patch my past traces with the latest model costs.

However I am getting this error:
Error: Cannot find module '/home/[user]/langfuse/web/node_modules/@langfuse/shared/dist/src/db.js'
code: 'MODULE_NOT_FOUND',
path: '/home/[user]/langfuse/web/node_modules/@langfuse/shared/package.json'

Expected: DB migration should take place seamlessly

### To reproduce

Run the exact same commands as mentioned in https://langfuse.com/docs/deployment/self-host#migrate-models

### SDK and container versions

Running latest version of langfuse

### Are you interested to contribute a fix for this bug?

No

**Comment by marcklingen** [comment]
thanks for reporting, fixed in #2703. Please pull again from main and try again

**Comment by atrik-das-razer** [comment]
Hi @marcklingen,

Thanks for your prompt response. I tried running the script again and this time I didn't face any dependency issues which was great but I did face this typing error:
![image](https://github.com/user-attachments/assets/1e993a0a-91f2-42fa-97b2-123bf4b3cdbc)

I fixed it by editing the file in web/node_modules/@langfuse/shared/src/features/ingestion/types.ts with the following code:
```
.superRefine((data, ctx) => {
  // Type guard to help TypeScript with type narrowing
    function isOfType<T extends { dataType?: string }>(d: T, type: string): d is T & { dataType: string } {
      return d.dataType === type;
    }

    if (isOfType(data, "CATEGORICAL") && typeof data.value === "number") {
      ctx.addIssue({
        code: z.ZodIssueCode.custom,
        message: `Value must be a string for data type ${data.dataType}`,
      });
    } else if (isOfType(data, "NUMERIC") && typeof data.value === "string") {
      ctx.addIssue({
        code: z.ZodIssueCode.custom,
        message: `Value must be a number for data type ${data.dataType}`,
      });
    } else if (isOfType(data, "BOOLEAN") && typeof data.value === "string") {
      ctx.addIssue({
        code: z.ZodIssueCode.custom,
        message: `Value must number equal to either 0 or 1 for data type ${data.dataType}`,
      });
    }
  });
```
After this change, the script ran just fine and seemed to have updated the historical token usage for all my traces however my cost is still showing as $0.00:
![image](https://github.com/user-attachments/assets/a5a09e09-f688-4c1a-a777-a779944d45c0)

Shouldn't the cost also be updated accordingly?
